soda-core/src/soda_core/contracts/contract_verification.py

```diff
@@ -297,25 +298,42 @@ def log_table_row_diagnostics(self, verbose: bool = True) -> str:
         diagnostics = []
 
         if self.diagnostic_metric_values:
-            for metric_name, value in self.diagnostic_metric_values.items():
-                formatted_value = value
-                if not verbose:
-                    formatted_value = self._log_console_format(value)
-                diagnostics.append(f"{metric_name}: {formatted_value}")
+            # If the diagnostic metric values implements the ISodaCloudOutput interface, use it
+            if isinstance(self.diagnostic_metric_values, ILoggingOutput):
```

Comment:
I wonder how this works for a list of ILoggingOutput instances, it would skip this part and go into `_log_console_format` where handling of that interface is not present, or am I mistaken? Wouldn't it make sense to handle ILoggingOutput in there alongside other types so that recursive calls are handled as well?